### ISSUE TITLE
Verbose search fails when expected metadata is missing

### ISSUE THREAD

**Issue by tkkuehn** [boutiques/boutiques#655]
TLDR: `bosh search -v` errors out right now.

When doing a verbose search, boutiques assumes every Zenodo record it finds will have a version specified under its metadata. If this is not the case, processing the search JSON will fail with a `KeyError`:

https://github.com/boutiques/boutiques/blob/5a2374e87dd837cb83fec9603f32e3c9085613ba/tools/python/boutiques/searcher.py#L121

In the last week or so, a couple of Boutiques tools have been published ([This one](https://zenodo.org/record/8357249) and [this one](https://zenodo.org/record/8365022), check the [JSON](https://zenodo.org/record/8357249/export/json) [export](https://zenodo.org/record/8365022/export/json) to confirm) that have no version defined in their metadata, causing a verbose search for all Boutiques tools to fail without returning anything.

Instead of failing, I think Boutiques should either:

1. Not require a version from tools and process them without.
2. Skip tools that don't have a version, possibly raising a warning.

I can go ahead and make either of these changes in a PR depending on what you think

**Comment by glatard** [comment]
Hi @tkkuehn, sorry for the late reply. I think the way to go would be to fallback on an "unknown" version if the version isn't present in the descriptor. So solution 1. Please go ahead and submit a PR, thanks a lot for reporting this!
